tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -392,8 +392,9 @@ uint32_t DeclResultIdMapper::createVarOfExplicitLayoutStruct(
     auto varType = declDecl->getType();
     varType.removeLocalConst();
 
+    const bool isRowMajor = declDecl->hasAttr<HLSLRowMajorAttr>() || !declDecl->hasAttr<HLSLColumnMajorAttr>() && spirvOptions.HLSLDefaultRowMajor;
```

Comment:
This is likely to cause problems for non-matrix types since we have an assert in `translateType()` to make sure `isRowMajor` is only set for (array of) matrices: https://github.com/Microsoft/DirectXShaderCompiler/blob/578b0df014c56cbf1aba631ad35e6ea13face358/tools/clang/lib/SPIRV/TypeTranslator.cpp#L111-L115

Besides, the same logic shows up again in `TypeTranslator`. What about having a public function in `TypeTranslator` to wrap around this logic and also check the type there?

Reply:
I was aware of the assert but kept it in still.
Yes, that sounds like a better approach? Should I try to remove the isRowMajor argument altogether?

Reply:
I don't think it's possible to remove `isRowMajor` from the signature of `translateType()` because `QualType` does not contain majorness information.